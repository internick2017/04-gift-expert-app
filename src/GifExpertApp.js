import React, { useState } from "react";


const GifExpertApp = () => {

    // const categories = ['One Punch', 'Samurai X', 'Dragon ball']

    const [categories, setCategories] = useState(['One Punch', 'Samurai X', 'Dragon ball']);

    const handleApp = () => {
        // setCategories([...categories, 'Shaman King']);
        // setCategories(['Shaman King', ...categories]);
        setCategories(cate => [...cate, 'Stein Gates']);
    }
    return (

        <>
            <h2> GifExpertApp </h2>
            <hr />

            <button onClick={handleApp}> Agregar </button>
            <ol>

                {
                    categories.map(category => {
                        return <li key={category}>{category}</li>
                    })

                }

            </ol>

        </>

    );

}

export default GifExpertApp;