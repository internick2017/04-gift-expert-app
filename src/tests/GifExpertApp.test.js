import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en GifExpertApp', () => {

    test('deve hacer match con el snapshot', () => {

        const wrapper = shallow(<GifExpertApp />)

        expect(wrapper).toMatchSnapshot();

    });

    test('deve de mostrar una lista de categorias', () => {
        const categories = ['Bleach', 'Death note'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    });

});
