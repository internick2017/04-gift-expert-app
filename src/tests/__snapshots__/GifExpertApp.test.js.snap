// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp deve de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Bleach"
      key="Bleach"
    />
    <GifGrid
      category="Death note"
      key="Death note"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en GifExpertApp deve hacer match con el snapshot 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
